src/queues.c

```diff
@@ -347,7 +347,10 @@ void queues_history_clear(void)
         if (g_queue_is_empty(history))
                 return;
 
-        g_queue_clear_full(history, (GDestroyNotify)notification_unref);
+        while (!g_queue_is_empty(history)) {
+                struct notification *n = g_queue_pop_head(history);
+                notification_unref(n);
+        }
```

Comment:
You can actually use the source of glib to copy the function here. See https://gitlab.gnome.org/GNOME/glib/-/blob/main/glib/gqueue.c#L166

```
g_queue_foreach (queue, (GFunc) free_func, NULL);
g_queue_clear (queue);
```